### PR TITLE
Adding shadows to design tokens

### PR DESCRIPTION
**Pull request by georgewrmarshall**
### Description 
Adding shadows to design tokens.

The main architecture decisions for shadow tokens is to split up the size(x,y,blur,spread) and colors(neutral, primary, error/danger)

This adds 3 new colors to the colors:
- shadow/default
- primary/shadow
- error/shadow

## Screenshots

shadow

## References
https://www.figma.com/file/kdFzEC7xzSNw7cXteqgzDW/DS-Colors-%26-Shadows-%2F-Light-Theme?node-id=753%3A801
https://www.figma.com/file/rLKsoqpjyoKauYnFDcBIbO/DS-Colors-%26-Shadows-%2F-Dark-Theme?node-id=522%3A1022

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import '../src/css/design-tokens.css';
+
 export const parameters = {
   actions: { argTypesRegex: '^on[A-Z].*' },
   controls: {
```
Comment (georgewrmarshall) [inline .storybook/preview.js:5]: Adding css variable stylesheet to storybook

File: src/js/colors/colors.ts [hunk 2]
```diff
@@ -106,6 +114,7 @@ export const colors: Colors = {
       muted: '#D73A4919',
       inverse: '#FCFCFC',
       disabled: '#D73A4980',
+      shadow: '#d73a4966',
```
Comment (Cal-L) [inline src/js/colors/colors.ts:117]: Tiniest nit pick - all caps ? :)

File: docs/Shadows.mdx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+| **Card**                 | `cardShadow: {...shadows.size.xs, colors.shadow.default }`         | `box-shadow: var(--shadow-size-xs) var(--color-shadow-default);` |
+| **Dropdown**             | `dropdownShadow: { ...shadows.size.sm, colors.shadow.default}`     | `box-shadow: var(--shadow-size-sm) var(--color-shadow-default);` |
+| **Toast**                | `toastShadow: { ...shadows.size.md, colors.shadow.default}`        | `box-shadow: var(--shadow-size-md) var(--color-shadow-default);` |
+| **Modal**                | `modalShadow: { ...shadows.size.lg, colors.shadow.default}`        | `box-shadow: var(--shadow-size-lg) var(--color-shadow-default);` |
+| **Button Primary Hover** | `buttonPrimaryHover: { ...shadows.size.sm, colors.primary.shadow}` | `box-shadow: var(--shadow-size-sm) var(--color-primary-shadow);` |
+| **Button Danger Hover**  | `buttonDangerHover: { ...shadows.size.sm, colors.primary.shadow}`  | `box-shadow: var(--shadow-size-sm) var(--color-error-shadow);`   |
+
+The CSS-in-JS `shadows.size` objects for React Native do contain a default color but colors should be overridden so the correct shadow colors are displayed in both light and dark themes.
```
Comment (Cal-L) [inline docs/Shadows.mdx:56]: Would it make sense to split shadows into a `light` and `dark` object as well and each would use shared shadow styles + replace colors accordingly? That way, the app wouldn't need to concern itself with replacing colors in addition to shadows since shadows will be handled the same as colors. WDYT?

File: src/js/colors/colors.ts (hunk 2 above)
Comment (georgewrmarshall) [reply src/js/colors/colors.ts:117]: Nice catch!

File: jest.config.js [hunk 4]
```diff
@@ -2,7 +2,11 @@ module.exports = {
   collectCoverage: true,
   // Ensures that we collect coverage from all source files, not just tested
   // ones.
-  collectCoverageFrom: ['./src/**/*.ts', '!./src/**/*index.ts'],
+  collectCoverageFrom: [
+    './src/**/*.ts',
+    '!./src/**/*index.ts',
+    '!./src/js/colors/colors.ts',
```
Comment (georgewrmarshall) [inline jest.config.js:8]: Ignoring coverage for the deprecated colors file as it just uses the objects from light and dark theme color which there are already tests for

File: src/js/colors/colors.test.ts [hunk 5]
```diff
@@ -1,452 +0,0 @@
...
-      designTokens.dark.colors.info.muted.value,
-    );
-  });
-
-  it('js tokens for info.inverse matches figma tokens info.inverse', () => {
-    expect(importableColors.dark.info.inverse).toStrictEqual(
-      designTokens.dark.colors.info.inverse.value,
-    );
-  });
-});
```
Comment (georgewrmarshall) [inline src/js/colors/colors.test.ts:452]: Removing this file so we don't have to keep 2 versions of the unit tests for colors and ignoring from jest config

File: docs/Shadows.mdx (hunk 3 above)
Comment (georgewrmarshall) [reply docs/Shadows.mdx]: Updated

**Review by Cal-L — DISMISSED** [review]
LGTM

File: docs/Shadows.mdx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+## Size
+
+There are 4 different sizes of shadow in MetaMask.
+
+<Canvas>
+  <Story id="shadows-shadows--size" />
+</Canvas>
+
+| Size   | JS                | CSS                     |
```
Comment (brianacnguyen) [inline docs/Shadows.mdx:18]: Do these token sizes and colors align in figma?

File: docs/Shadows.mdx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  <Story id="shadows-shadows--color" />
+</Canvas>
+
+| Color       | JS                      | CSS                           |
+| ----------- | ----------------------- | ----------------------------- |
+| **neutral** | `colors.shadow.default` | `var(--color-shadow-default)` |
+| **primary** | `colors.primary.shadow` | `var(--color-primary-shadow)` |
+| **danger**  | `colors.error.shadow`   | `var(--color-error-shadow)`   |
+
+## Usage
```
Comment (brianacnguyen) [inline docs/Shadows.mdx:39]: maybe change to "Sample Usage". from my understanding these shadows can be used outside of the components listed. simple "Usage" may suggest exclusivity

File: docs/Shadows.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+          }}
+        >
+          <Text as="p" style={{ margin: 0 }}>
+            Button Danger Hover
+          </Text>
+        </ShadowSwatch>
+      </div>
+    </div>
+  );
+};
```
Comment (brianacnguyen) [inline docs/Shadows.stories.tsx:206]: Not sure if we have a standard practice for storybook yet, but i personally think a story with knobs with shadow colors and sizes will be beneficial here

File: docs/components/Text/Text.tsx [hunk 9]
```diff
@@ -1,4 +1,5 @@
 import React from 'react';
```
Comment (brianacnguyen) [inline docs/components/Text/Text.tsx:1]: can you explain why this change was included in this cl? if it's a systematic change it'd be better to make a separate cl for it

File: docs/Shadows.mdx (hunk 6 above)
Comment (georgewrmarshall) [reply docs/Shadows.mdx:18]: Great question these should align with the tokens in Figma yes. I think Figma library is currently out of date @Akatori-Design

File: docs/Shadows.mdx (hunk 7 above)
Comment (georgewrmarshall) [reply docs/Shadows.mdx]: Resolved https://github.com/MetaMask/design-tokens/pull/137/commits/c55bf5a671de6c4ce3780c7d3ce0ccfac02431d0

File: docs/Shadows.stories.tsx (hunk 8 above)
Comment (georgewrmarshall) [reply docs/Shadows.stories.tsx:238]: I usually save controls for component stories with props but I suppose it could add some value here.  Resolved in https://github.com/MetaMask/design-tokens/pull/137/commits/c55bf5a671de6c4ce3780c7d3ce0ccfac02431d0

File: docs/components/Text/Text.tsx (hunk 9 above)
Comment (georgewrmarshall) [reply docs/components/Text/Text.tsx:1]: This update make Euclid our brand font the default font family for this Text component. All components in this repo are used to just helo the documentation and examples of the design tokens that should be the main focus.

**Review by brianacnguyen — APPROVED** [review]

**Review by Cal-L — APPROVED** [review]
LGTM
